Makefile

```diff
@@ -50,95 +59,179 @@ e2e-test:
+##################################################	
 .PHONY: build
-build: gofmt checkenv build-adapter build-controller
-
-.PHONY: build-controller
-build-controller: generate-api pkg/scalers/liiklus/LiiklusService.pb.go
-	$(GO_BUILD_VARS) operator-sdk build $(IMAGE_CONTROLLER) \
-		--go-build-args "-ldflags -X=github.com/kedacore/keda/version.Version=$(VERSION) -o build/_output/bin/keda"
-
-.PHONY: build-adapter
-build-adapter: generate-api pkg/scalers/liiklus/LiiklusService.pb.go
-	$(GO_BUILD_VARS) go build \
-		-ldflags "-X=github.com/kedacore/keda/version.GitCommit=$(GIT_COMMIT) -X=github.com/kedacore/keda/version.Version=$(VERSION)" \
-		-o build/_output/bin/keda-adapter \
-		cmd/adapter/main.go
-	docker build -f build/Dockerfile.adapter -t $(IMAGE_ADAPTER) .
-
-.PHONY: generate-api
-generate-api:
-	$(GO_BUILD_VARS) operator-sdk generate k8s
-	$(GO_BUILD_VARS) operator-sdk generate crds --crd-version=v1beta1
-	# withTriggers and withPods are only used for duck typing so we only need the deepcopy methods
+build: manifests set-version manager adapter
+
+# Build the docker image
+docker-build: build
+	docker build . -t ${IMAGE_CONTROLLER}
+	docker build -f Dockerfile.adapter -t ${IMAGE_ADAPTER} .
+
+# Build KEDA Operator binary
+.PHONY: manager
+manager: generate gofmt govet pkg/scalers/liiklus/LiiklusService.pb.go
+	${GO_BUILD_VARS} go build \
+	-ldflags "-X=github.com/kedacore/keda/version.GitCommit=$(GIT_COMMIT) -X=github.com/kedacore/keda/version.Version=$(VERSION)" \
+	-o bin/keda main.go
+
+# Build KEDA Metrics Server Adapter binary
+.PHONY: adapter
+adapter: generate gofmt govet pkg/scalers/liiklus/LiiklusService.pb.go
+	${GO_BUILD_VARS} go build \
+	-ldflags "-X=github.com/kedacore/keda/version.GitCommit=$(GIT_COMMIT) -X=github.com/kedacore/keda/version.Version=$(VERSION)" \
+	-o bin/keda-adapter adapter/main.go
+
+# Generate manifests e.g. CRD, RBAC etc.
+.PHONY: manifests
+manifests: controller-gen
+	$(CONTROLLER_GEN) crd:crdVersions=v1beta1 rbac:roleName=keda-operator paths="./..." output:crd:artifacts:config=config/crd/bases
+	# withTriggers is only used for duck typing so we only need the deepcopy methods
 	# However operator-sdk generate doesn't appear to have an option for that
 	# until this issue is fixed: https://github.com/kubernetes-sigs/controller-tools/issues/398
-	rm deploy/crds/keda.sh_withtriggers_crd.yaml
-	rm deploy/crds/keda.sh_withpods_crd.yaml
+	rm config/crd/bases/keda.sh_withtriggers.yaml
+
+# Generate code (API)
+.PHONY: generate
+generate: controller-gen
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+# find or download controller-gen
+# download controller-gen if necessary
+.PHONY: controller-gen
+controller-gen:
+ifeq (, $(shell which controller-gen))
+	@{ \
+	set -e ;\
+	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
+	cd $$CONTROLLER_GEN_TMP_DIR ;\
+	go mod init tmp ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 ;\
+	rm -rf $$CONTROLLER_GEN_TMP_DIR ;\
+	}
```

Comment:
just curious if 

```
GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0
```

could be used here, or if this tmp directory is needed.

Reply:
Good catch, you are right. This part of the Makefile is directly from the Kubebuilder scaffolding. I have changed it as you suggested, if there's any problem in the future, we can revert back.